lib/react-dom.js

```diff
@@ -404,19 +404,281 @@ declare type $JSXIntrinsics = {
+  onWheel?: (SyntheticWheelEvent<E>) => void,
+  onWheelCapture?: (SyntheticWheelEvent<E>) => void,
+};
+
+type ReactDOM$ElementProps = {
+  children?: React$Node,
+  className?: string,
+  dangerouslySetInnerHTML?: { __html: string },
+  id?: string,
+  lang?: string,
+  style?: ReactDOM$Style,
+  suppressHydrationWarning?: boolean,
+  tabIndex?: ReactDOM$Number,
+
+  // NOTE: There is no way we can enumerate all acceptable/standard props of DOM
+  // intrinsics. If we ever got a Flow feature allowing regexes for keys, we
+  // could express open-ended namespaces like data-* attributes, custom
+  // attributes etc more precisely, and then maybe think about making this
+  // definition complete/exact (thus detecting more classes of errors).
+  //
+  // For now the typing philosophy here is: be reasonably strict with the types
+  // of known attributes, but lenient otherwise.
+
+  [key: string]: any
+};
+
+type ReactDOM$HTMLElementProps = ReactDOM$ElementProps & {
+  accessKey?: string,
+  autoCapitalize?: $TODO$CaseInsensitive<"off" | "none" | "on" | "sentences" | "words" | "characters">,
+  contentEditable?: ReactDOM$BooleanishString,
+  contextMenu?: string,
+  dir?: $TODO$CaseInsensitive<"ltr" | "rtl" | "LTR" | "RTL" | "auto">,
+  draggable?: ReactDOM$BooleanishString,
+  hidden?: ReactDOM$Boolean<"hidden">,
+  is?: string,
+  itemProp?: string,
+  itemRef?: string,
+  itemScope?: ReactDOM$Boolean<"itemScope" | "itemscope">,
+  itemType?: string,
+  slot?: string,
+  spellCheck?: ReactDOM$BooleanishString,
+  suppressContentEditableWarning?: boolean,
+  title?: string,
+  translate?: $TODO$CaseInsensitive<"yes" | "no">
+};
+
+type ReactDOM$SVGElementProps = ReactDOM$ElementProps & {
+  xlinkActuate?: string,
+  xlinkArcrole?: string,
+  xlinkHref?: string,
+  xlinkRole?: string,
+  xlinkShow?: string,
+  xlinkTitle?: string,
+  xlinkType?: string,
+  xmlBase?: string,
+  xmlLang?: string,
+  xmlSpace?: string
+
+  // TODO: SVG presentation props
+  // TODO: SVG filters props
+  // TODO: SVG animation props
+  // TODO: SVG event props
+};
+
+type ReactDOM$BooleanishString =
+  | $TODO$CaseInsensitive<"true" | "false">
+  | boolean;
+
+type ReactDOM$Boolean<AttributeName: string> =
+  | $TODO$CaseInsensitive<AttributeName>
+  | boolean;
+
+type ReactDOM$HTMLElementJSXIntrinsic<Instance = HTMLElement, Props = ReactDOM$HTMLElementProps> = {
+  instance: Instance,
+  props: Props & ReactDOM$GlobalEventHandlers<Instance>,
 };
 
 type ReactDOM$SVGElementJSXIntrinsic = {
   instance: Element,
-  props: {children?: React$Node, [key: string]: any},
+  props: ReactDOM$SVGElementProps,
+};
+
+// TODO: Can we get an actual Flow $CaseInsensitive builtin for singleton
+// strings and enums?
+type $TODO$CaseInsensitive<Value: string> = Value;
+
+type ReactDOM$Style = Object; // TODO
+
+// NOTE: In principle this can be `number | string`, but constraining values to
+// numbers might be a useful design choice.
+type ReactDOM$Number = number;
```

Comment:
This causes quite a few errors might wanna consider making it a `number | string`.

Reply:
made it `number | string`